packages/server-wallet/src/protocols/__test__/open-channel.test.ts

```diff
@@ -1,95 +1,210 @@
+      await testCase({participant: 1, statesHeld: [1], statesToSign: []});
+    });
+    it(`signs the pre-fund-setup if I have my opponent's state`, async () => {
+      await testCase({participant: 1, statesHeld: [0], statesToSign: [1]});
+    });
+  });
 });
 
-const fundChannelAction2 = {...fundChannelAction1, expectedHeld: BN.from(5)};
-
-test.each`
-  supported           | myIndex | latestSignedByMe    | latest              | funding      | action                         | cond                                                                     | result
-  ${undefined}        | ${0}    | ${undefined}        | ${prefundState1}    | ${notFunded} | ${signState(prefundState0)}    | ${'when the prefund state is not supported, and I am index 0'}           | ${'signs the prefund state'}
-  ${undefined}        | ${0}    | ${undefined}        | ${reversedPFState1} | ${notFunded} | ${signState(reversedPFState0)} | ${'when the prefund state is not supported, and I am index 1'}           | ${'signs the prefund state'}
-  ${prefundState0}    | ${0}    | ${prefundState0}    | ${prefundState0}    | ${funded}    | ${signState(postfundState2)}   | ${'when the prefund state is supported, and the channel is funded'}      | ${'signs the prefund state'}
-  ${prefundState1}    | ${0}    | ${prefundState0}    | ${prefundState1}    | ${funded}    | ${signState(postfundState2)}   | ${'when the prefund state is supported, and the channel is funded'}      | ${'signs the postfund state'}
-  ${prefundState1}    | ${1}    | ${prefundState1}    | ${prefundState1}    | ${funded}    | ${signState(postfundState3)}   | ${'when the prefund state is supported, and the channel is funded'}      | ${'signs the postfund state'}
-  ${prefundState0}    | ${0}    | ${prefundState0}    | ${prefundState0}    | ${notFunded} | ${fundChannelAction1}          | ${'when the prefund state is supported, and alice is first to deposit'}  | ${'submits transaction'}
-  ${altPrefundState0} | ${0}    | ${altPrefundState0} | ${altPrefundState0} | ${funded}    | ${fundChannelAction2}          | ${'when the prefund state is supported, and alice is second to deposit'} | ${'submits transaction'}
-`('$result $cond', ({supported, latest, latestSignedByMe, funding, action, myIndex}) => {
-  const directFundingStatus = directFundingStatusFn(
-    supported,
-    funding,
-    participants[myIndex],
-    'Direct'
-  );
-  const ps = applicationProtocolState({
-    app: {
-      supported,
-      latest,
-      latestSignedByMe,
-      funding,
-      myIndex,
-      directFundingStatus,
-    },
+describe(`funding phase`, () => {
+  const statesHeld = [0, 1];
+
+  describe(`as participant 0`, () => {
+    it(`submits the transaction if it is my turn`, async () => {
+      const o = await testCase({participant: 0, statesHeld, totalFunds: 0, fundsToDeposit: 5});
+      // and then doesn't submit it a second time
+      await testCrankOutcome(o, {fundsToDeposit: 0});
```

Comment:
Looks like `fundsToDeposit` defaults to `0`, do you need to specify it again here?

It's not immediately obvious that awaiting `testCrankOutcome` with a certain input is equivalent to asserting that a transaction is not submitted. a second time.

Reply:
I wanted to make it more obvious what I was checking for here. You're right though that it still isn't very obvious. Maybe `crankAndAssert` would be a better name?